src/cryptsetup/cryptsetup.c

```diff
@@ -1492,7 +1492,7 @@ static int attach_luks_or_plain_or_bitlk_by_pkcs11(
                                 return r;
 
                         log_notice("Security token %s not present for unlocking volume %s, please plug it in.",
-                                   uri, friendly);
+                                   strempty(uri), friendly);
```

Comment:
Hmm, when `uri` is NULL, the log contains doubled space. I'm not familiar with the logic, but maybe we should conditinoalize the message with if uri is NULL?

Reply:
The double space doesn't seem so ugly to me, but yes, we can improve the message if you wish.